tests/flytekit/unit/clients/auth/test_authenticator.py

```diff
@@ -77,13 +84,11 @@ def test_client_creds_authenticator(mock_requests):
         http_proxy_url="https://my-proxy:31111",
     )
 
-    response = MagicMock()
-    response.status_code = 200
-    response.json.return_value = json.loads("""{"access_token": "abc", "expires_in": 60}""")
-    mock_requests.post.return_value = response
     authn.refresh_credentials()
     expected_scopes = static_cfg_store.get_client_config().scopes
+
     assert authn._creds
```

Comment:
The order of mocking vs instantiation of the authenticator matters now.
I added the more explicit check `assert authn._creds.access_token == "abc"` because this would fail in case somebody in the future reverses the order again (instantiate the authenticator first, then mock).